crates/ibc/src/core/ics03_connection/version.rs

```diff
@@ -32,9 +32,33 @@ pub struct Version {
 }
 
 impl Version {
+    /// Check whether or not the given version is compatible with supported versions
+    pub fn verify_version_supported(
+        &self,
+        supported_versions: &[Version],
+    ) -> Result<(), ConnectionError> {
+        let maybe_version_supported = supported_versions
+            .iter()
+            .find(|sv| sv.identifier == self.identifier)
+            .ok_or(ConnectionError::VersionNotSupported {
+                version: self.clone(),
+            })?;
+
+        for feature in self.features.iter() {
+            maybe_version_supported.verify_feature_supported(feature.to_string())?;
+        }
```

Comment:
ibc-go also [ensures](https://github.com/cosmos/ibc-go/blob/295bab01f633633e0d6136ba9b974c0e248edbe5/modules/core/03-connection/types/version.go#L80) that there is at least 1 feature.

Reply:
[fdab426](https://github.com/cosmos/ibc-rs/pull/626/commits/fdab426831e6f160d5646ad19c1349ed26b4021b)